drivers/metal/rendering_device_driver_metal.mm

```diff
@@ -0,0 +1,3919 @@
+	MTLSize clamped;
+	clamped.width = MIN(p_size.width, p_max_size.width - p_origin.x);
+	clamped.height = MIN(p_size.height, p_max_size.height - p_origin.y);
+	clamped.depth = MIN(p_size.depth, p_max_size.depth - p_origin.z);
+	return clamped;
+}
+
+void RenderingDeviceDriverMetal::command_copy_texture(CommandBufferID p_cmd_buffer, TextureID p_src_texture, TextureLayout p_src_texture_layout, TextureID p_dst_texture, TextureLayout p_dst_texture_layout, VectorView<TextureCopyRegion> p_regions) {
+	MDCommandBuffer *cmd = (MDCommandBuffer *)(p_cmd_buffer.id);
+	id<MTLTexture> src = rid::get(p_src_texture);
+	id<MTLTexture> dst = rid::get(p_dst_texture);
+
+	id<MTLBlitCommandEncoder> blit = cmd->blit_command_encoder();
+	PixelFormats &pf = *pixel_formats;
+
+	MTLPixelFormat src_fmt = src.pixelFormat;
+	bool src_is_compressed = pf.getFormatType(src_fmt) == MTLFormatType::Compressed;
+	MTLPixelFormat dst_fmt = dst.pixelFormat;
+	bool dst_is_compressed = pf.getFormatType(dst_fmt) == MTLFormatType::Compressed;
+
+	// validate copy
+	if (src.sampleCount != dst.sampleCount || pf.getBytesPerBlock(src_fmt) != pf.getBytesPerBlock(dst_fmt)) {
+		ERR_FAIL_MSG("Cannot copy between incompatible pixel formats, such as formats of different pixel sizes, or between images with different sample counts.");
+	}
+
+	// If source and destination have different formats and at least one is compressed, a temporary buffer is required
+	bool need_tmp_buffer = (src_fmt != dst_fmt) && (src_is_compressed || dst_is_compressed);
+	if (need_tmp_buffer) {
+		ERR_FAIL_MSG("not implemented: copy with intermediate buffer");
+	}
+
+	if (src_fmt != dst_fmt) {
+		// Map the source pixel format to the dst through a texture view on the source texture.
+		src = [src newTextureViewWithPixelFormat:dst_fmt];
+	}
+
+	for (uint32_t i = 0; i < p_regions.size(); i++) {
+		TextureCopyRegion region = p_regions[i];
+
+		MTLSize extent = MTLSizeFromVector3i(region.size);
+
+		// If copies can be performed using direct texture-texture copying, do so
+		uint32_t src_level = region.src_subresources.mipmap;
+		uint32_t src_base_layer = region.src_subresources.base_layer;
+		MTLSize src_extent = mipmapLevelSizeFromTexture(src, src_level);
+		uint32_t dst_level = region.dst_subresources.mipmap;
+		uint32_t dst_base_layer = region.dst_subresources.base_layer;
+		MTLSize dst_extent = mipmapLevelSizeFromTexture(dst, dst_level);
+
+		// All layers may be copied at once, if the extent completely covers both images.
+		if (src_extent == extent && dst_extent == extent) {
+			[blit copyFromTexture:src
+						 sourceSlice:src_base_layer
+						 sourceLevel:src_level
+						   toTexture:dst
+					destinationSlice:dst_base_layer
+					destinationLevel:dst_level
+						  sliceCount:region.src_subresources.layer_count
+						  levelCount:1];
+		} else {
+			MTLOrigin src_origin = MTLOriginFromVector3i(region.src_offset);
+			MTLSize src_size = clampMTLSize(extent, src_origin, src_extent);
+			uint32_t layer_count;
```

Comment:
```suggestion
			uint32_t layer_count = 0;
```

The current guideline in Godot is to initialize everything, assuming most redundant initializations will generally be removed by the compiler and they help with debugging.

Reply:
Fixed